src/main/java/org/tugraz/sysds/hops/DataGenOp.java

```diff
@@ -454,8 +457,8 @@ public boolean hasConstantValue(double val)
 		//sparsity awareness if requires
 		if( ret && val != 0 ) {
 			Hop sparsity = getInput().get(_paramIndexMap.get(DataExpression.RAND_SPARSITY)); //sparsity
-			ret &= (sparsity == null || sparsity instanceof LiteralOp 
-				&& HopRewriteUtils.getDoubleValueSafe((LiteralOp)sparsity) == 1);
+			ret = (sparsity == null || sparsity instanceof LiteralOp
+					&& HopRewriteUtils.getDoubleValueSafe((LiteralOp) sparsity) == 1);
```

Comment:
This does not respect the previously computed condition if min and max are equal

Reply:
ok never mind - its covered under the condition.